src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,842 @@
+                    value = 0;
+                    return false;
+                }
+            }
+
+            // create the new AdcValue object and calculate the voltage
+            var adcValue = new AdcValue() { Raw = raw, Time = _stopWatch.ElapsedTicks, Channel = (byte)(GetRegisterBits(Register.Status, BitMask.StatusCHD)), Voltage = RawValueToVoltage(raw) };
+
+            // add it to the collection
+            _adcValues.Add(adcValue);
+
+            // call the event handler
+            OnValueReceived?.Invoke(this, new AdcValueReceivedEventArgs(adcValue));
+
+            value = raw;
+
+            return true;
+        }
+
+        /// <summary>
+        /// Selects the channel, starts a single conversion, waits for its completion and returns the read value without converting it to Volts
+        /// </summary>
+        /// <param name="channel">Channel index</param>
+        /// <returns>24-bit raw value of the ADC result</returns>
+        public uint? ReadSingleADCValue(Channel channel)
+        {
+            ActiveChannels = channel;
+
+            StartSingleConversion();
+
+            WaitForADC();
+
+            return TryReadADCValue(out uint rawValue) ? rawValue : (uint?)null;
+        }
+
+        /// <summary>
+        /// Converts the raw ADC result to Volts
+        /// </summary>
+        /// <param name="adcValue">The raw ADC result</param>
+        /// <returns>The ADC result value in Volts</returns>
+        private double RawValueToVoltage(uint adcValue)
+        {
+            int pgaGain = 1;
+            switch (PGAGain)
+            {
+                case Gain.X1:
+                    pgaGain = 1;
+                    break;
+                case Gain.X8:
+                    pgaGain = 8;
+                    break;
+                case Gain.X16:
+                    pgaGain = 16;
+                    break;
+                case Gain.X32:
+                    pgaGain = 32;
+                    break;
+                case Gain.X64:
+                    pgaGain = 64;
+                    break;
+                case Gain.X128:
+                    pgaGain = 128;
+                    break;
+            }
+
+            double voltage = 0;
+
+            // 0 - bipolar (ranges from ±19.53 mV to ±2.5 V) ; 1 - unipolar (ranges from 0 mV to 19.53 mV to 0 V to 2.5 V)
+            if (Unipolar)
+            {
+                voltage = (double)adcValue / 16777216.0;
+            }
+            else if (!Unipolar)
+            {
+                voltage = ((double)adcValue / 8388608.0) - 1.0;
+            }
+
+            voltage *= VoltageReference / pgaGain;
+
+            return voltage;
+        }
+
+        /// <summary>
+        /// Converts the ADC result to Celsius
+        /// </summary>
+        /// <param name="adcValue">The raw ADC result</param>
+        /// <returns></returns>
+        protected Temperature ADCValueToTemperature(ulong adcValue)
```

Comment:
if this is for analog thermometers then please remove the helper. We should decide on the helpers once we got abstraction in place

Reply:
The thermometer is part of the AD7193 IC.